service/controller/azuremachinepool/handler/migration/deleteold.go

```diff
@@ -0,0 +1,27 @@
+package migration
+
+import (
+	"context"
+
+	oldcapzexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capzexp/v1alpha3"
+	"github.com/giantswarm/microerror"
+)
+
+func (r *Resource) deleteOldAzureMachinePool(ctx context.Context, oldAzureMachinePool *oldcapzexpv1alpha3.AzureMachinePool) error {
```

Comment:
all other functions are in `resource.go` , why have this one on its own file?

Reply:
Also, I see we do some logic on the other `deleteOld*` function to handle the operatorkit finalizer. Don't we need that here?

Reply:
> all other functions are in resource.go , why have this one on its own file?

Good point, I was shuffling code across files, doesn't look pretty now. I'll do some refactoring to organize it better and more consistently.

Reply:
> Also, I see we do some logic on the other deleteOld* function to handle the operatorkit finalizer. Don't we need that here?

`machinepool` controller has the additional logic because of the following:
- We also have `machinepoolexp` controller which will reconcile the deletion.
- The above controller will, during deletion, remove the operatorkit finalizer for the old exp MachinePool `operatorkit.giantswarm.io/azure-operator-machine-pool-exp-controller`.
  - If this finalizer is not present, operatorkit returns an error (noticed this while testing).
  - So we want to keep this finalizer.
- Other controllers / operators might have set their own finalizers.
  - These finalizers will block CR deletion.
  - But we now have a new MachinePool CR, and everything should use the new one.
  - So we remove all finalizers in order not to block the deletion of the old MachinePool CR.

On the other hand, we don't have `azuremachinepoolexp` controller that is reconciling old AzureMachinePool CR, so we just remove all finalizers.